app/controllers/products_controller.rb

```diff
@@ -1,18 +1,67 @@
 class ProductsController < ApplicationController
+  before_filter :authenticate_user!, except: [ :index, :show ]
+
   def index
     @products = Product.all
     @categories = Category.all
   end
   def show
-        @categories = Category.all
-        @product = Product.find(params[:id])
-        @comments = Comment.where(product_id:params[:id])
+    @categories = Category.all
+    @product = Product.find(params[:id])
+    @designer = User.find(@product.designer_id)
+    @comments = Comment.where(product_id:params[:id])
+    @retailprice = @product.product_details.order("retail_price ASC").first
   end
   def new
+    @product = Product.new
   end
-
   def create
-    # ajout if logged in apres
-   Product.create!(name:params[:name],description:params[:description],image:params[:image])
+    @product = Product.new(name:params[:name],description:params[:description],image:params[:image],designer_id:current_user.id)
```

Comment:
Line is too long. [128/80]<br>Space missing after colon.<br>Space missing after comma.